Dockerfile.updater-core

```diff
@@ -95,13 +95,20 @@ COPY --from=ruby:3.1.4 --chown=dependabot:dependabot /usr/local /usr/local
 # This way other projects that import this library don't have to futz around with installing new / unexpected bundler versions.
 ARG BUNDLER_V2_VERSION=2.4.11
 
+ARG LIBYAML_VERSION=0.2.5
+RUN curl -sL https://pyyaml.org/download/libyaml/yaml-$LIBYAML_VERSION.tar.gz -o libyaml.tar.gz && \
+  mkdir -p $DEPENDABOT_HOME/src/libyaml && \
+  tar -xvf libyaml.tar.gz -C $DEPENDABOT_HOME/src/libyaml && \
+  rm libyaml.tar.gz
+
 RUN gem install bundler -v $BUNDLER_V2_VERSION --no-document && \
  rm -rf /var/lib/gems/*/cache/* && \
+ bundle config set --global build.psych --with-libyaml-source-dir=$DEPENDABOT_HOME/src/libyaml/yaml-$LIBYAML_VERSION && \
  bundle config set --local path 'vendor' && \
  bundle config set --local frozen 'true' && \
  bundle config set --local without 'development' && \
  bundle install && \
- rm -rf ~/.bundle
+ rm -rf ~/.bundle/cache
```

Comment:
Should this change happen regardless of the libyaml bump? Or is does somehow nuking the entire `~/.bundle` dir help with docker image caching more than nuking `~/.bundle/cache`?

Reply:
It's because I need to configure Bundler globally to use the downloaded sources and the previous `rm -rf ~/.bundle` would wipe out the configuration, which is also needed when installing gems from `omnibus/`.